app/react/Metadata/components/specs/SearchEntities.spec.tsx

```diff
@@ -0,0 +1,56 @@
+import { shallow, ShallowWrapper } from 'enzyme';
+import React from 'react';
+import { SearchEntities, SearchEntitiesProps, SearchEntitiesState } from '../SearchEntities';
+import SearchInput from 'app/Layout/SearchInput';
+import { sleep } from 'shared/tsUtils';
+import api from 'app/utils/api';
+
+import SearchResults from 'app/Connections/components/SearchResults';
+
+describe('SearchEntities', () => {
+  let component: ShallowWrapper<SearchEntitiesProps, SearchEntitiesState, SearchEntities>;
+  let props: SearchEntitiesProps;
+
+  beforeEach(() => {
+    props = {
+      onSelect: jasmine.createSpy('onSelect'),
+    };
+    spyOn(api, 'get').and.returnValue(
+      Promise.resolve({
+        json: {
+          rows: [
+            {
+              title: 'test',
+            },
+          ],
+        },
+      })
+    );
+  });
+
+  const render = () => {
+    component = shallow(<SearchEntities {...props} />);
+  };
+
+  describe('search', () => {
+    it('should request for the entities matching by title', async () => {
+      render();
+      component.find(SearchInput).simulate('change', { target: { value: 'test' } });
+      await sleep(4001);
```

Comment:
You know this better than me, but is this number arbitrary?

Reply:
I've changed it to 401, but it is related to the debounce time that is in the search function that has 400ms

Reply:
I've changed it to 401ms, but is related to the debounce time that the search function has.